src/Subtitles/index.js

```diff
@@ -0,0 +1,163 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 Metrological
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SubtitleAsset from './SubtitleAsset.js'
+export default class Subtitles {
+  // @ params url: subtitle file URL
+  // @return parsed subtitles as list of objects
+  // also stores parsed data
+  static fetchAndParseSubs(url) {
+    return fetch(url)
+      .then(data => data.text())
+      .then(subtitleData => {
+        this.clearCurrentSubtitle()
+        return this.parseSubtitles(subtitleData)
+      })
```

Comment:
I reckon this should contain a `.catch` block that handles the case where the `fetch` fails.